### ISSUE TITLE
Logger.warn/1 is deprecated on Elixir 1.15 

### ISSUE THREAD

**Issue by kipcole9** [kbrw/sweet_xml#93]
`Logger.warn/1` is deprecated on Elixir 1.15 and should (conditionally ?) be replaced with `Logger.warning/2`

```elixir
==> sweet_xml
Compiling 2 files (.ex)
warning: Logger.warn/1 is deprecated. Use Logger.warning/2 instead
  (sweet_xml 0.7.3) lib/sweet_xml/options.ex:107: SweetXml.Options.set_up/2
```

**Comment by thbar** [comment]
I've looked into this on various projects. Elixir 1.11 is the oldest maintained version of Elixir (https://hexdocs.pm/elixir/1.15/compatibility-and-deprecations.html) and `Logger.warning/2` was introduced exactly in that version 1.11.

If we require 1.11+ on this project, then there would be no need to conditionally call one or another.

This is the approach taken by:
- https://github.com/elixir-ecto/ecto/pull/4212

Alternatively, it can be done conditionally like this:
- https://github.com/elixir-ecto/ecto/pull/4215

I would personally just require 1.11+ and move on.

**Comment by thbar** [comment]
@kipcole9 I've proposed a change in #94, unsure if @Shakadak is still around to analyse the changes :smile:

**Comment by thbar** [comment]
@Shakadak many thanks! Is there someone else with commit rights on this repo?

**Comment by Shakadak** [comment]
Yes, there are a few, but the majority of them aren't displayed publicly in the organization members

I'll close this issue when we release the next version on hex.pm

**Comment by thbar** [comment]
#94 has been merged!

@Shakadak any chance you could issue a release for Elixir 1.15? Thanks!

**Comment by Shakadak** [comment]
Thanks for the reminder, it's published as [0.7.4](https://hex.pm/packages/sweet_xml/0.7.4)

**Comment by thbar** [comment]
Thank you!
